client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -61,8 +70,15 @@ export function updateMedia( action ) {
 	];
 }
 
-export const updateMediaSuccess = ( { siteId }, mediaItem ) => ( dispatch ) => {
+export const updateMediaSuccess = ( { siteId }, mediaItem ) => ( dispatch, getState ) => {
 	dispatch( receiveMedia( siteId, mediaItem ) );
+
+	if ( isEditorIframeLoaded( getState() ) ) {
```

Comment:
Let's take the following two things in consideration: 
* When an item has been both updated and removed, we're doing the same thing (sending a message to the iframe)
* Data layer handlers are generally supposed to create and dispatch actions, and here we're introducing a side effect

That being said, I'd recommend dispatching a new type of action here. Then, we could build a custom state middleware that would send the message when this action is dispatched. That will help keep the data layer handler clean of side effects, but would also help with the reusability of the code and would help to separate the concerns.

What do you think?

Reply:
this was partly adressed in https://github.com/Automattic/wp-calypso/commit/28ccf690cba9a2f11b85a9e43c27e31df60d9045 – as with other media actions I created a new thunk `gutenframeUpdateImageBlocks` (naming tbd) which takes care of the updates. As noted in https://github.com/Automattic/wp-calypso/pull/46483#discussion_r507628928, let me know if that approach would work for you